nixos/modules/system/boot/networkd.nix

```diff
@@ -1627,34 +1627,21 @@ let
 
   };
 
-
-  l2tpSessionOptions = {
-    options = {
-      l2tpSessionConfig = mkOption {
-        default = {};
-        type = types.addCheck (types.attrsOf unitOption) check.netdev.sectionL2TPSession;
-        description = ''
-          Each attribute in this set specifies an option in the
-          `[L2TPSession]` section of the unit.  See
-          {manpage}`systemd.netdev(5)` for details.
-        '';
-      };
-    };
-  };
-
-  wireguardPeerOptions = {
-    options = {
-      wireguardPeerConfig = mkOption {
-        default = {};
-        type = types.addCheck (types.attrsOf unitOption) check.netdev.sectionWireGuardPeer;
-        description = ''
-          Each attribute in this set specifies an option in the
-          `[WireGuardPeer]` section of the unit.  See
-          {manpage}`systemd.netdev(5)` for details.
-        '';
-      };
+  mkSubsectionType = oldKey: checkF:
```

Comment:
(bikeshedding potential, feel free to dismiss) should this be called mkSectionType? systemd's config format has sections but no "subsections", as far as I understand.

Reply:
I called it subsection because it's one level deeper than matchConfig/networkConfig/... in the Nix abstraction.
That said, I wouldn't worry too much since I plan to rip this out with my bare hands right after the 24.11 branch-off.